resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -705,6 +705,19 @@ private[spark] class Client(
       }
     }
 
+    val confDir =
+      sys.env.getOrElse("SPARK_CONF_DIR", sys.env("SPARK_HOME") + File.separator + "conf")
+    val dir = new File(confDir)
+    if (dir.isDirectory) {
+      val files = dir.listFiles(new FileFilter {
+        override def accept(pathname: File): Boolean = {
+          pathname.isFile && pathname.getName.endsWith("xml")
+        }
+      })
+      files.foreach { f => hadoopConfFiles(f.getName) = f }
```

Comment:
This indicates files in SPARK_CONF_DIR have higher priority than HADOOP_CONF_DIR or `YARN_CONF_DIR`, is it expected?

Reply:
yes, we follow that order to build classpath. plz check https://github.com/apache/spark/blob/12ab7f7e89ec9e102859ab3b710815d3058a2e8d/launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java#L134